src/app/courses/courses.component.ts

```diff
@@ -152,4 +162,53 @@ export class CoursesComponent implements OnInit, AfterViewInit {
     this.courses.data.forEach(row => this.selection.select(row));
   }
 
+  onFilterChange(filterValue: string, field: string) {
+    this.courses.filterPredicate = filterDropdowns(this.filter);
+    this.filter[field] = filterValue === 'All' ? '' : filterValue;
+    this.courses.filter = filterValue;
+  }
+
+  AdvanceSearchForm() {
+    const title = 'Advance Search';
+    const fields = this.AdvanceSearchFormFields();
+    const formGroup = this.AdvanceSearchFormGroup();
+    this.dialogsFormService
+      .confirm(title, fields, formGroup)
+      .subscribe((res) => {
+        if (res !== undefined) {
+          this.advanceSearchResponseDetails = res;
+          this.onFilterChange(this.advanceSearchResponseDetails.gradetext, this.advanceSearchResponseDetails.gradeLevel);
```

Comment:
Instead of calling the same function two times. please read all the form data and then send these parameters for filter.